src/main/java/com/dougnoel/sentinel/pages/PageManager.java

```diff
@@ -165,88 +158,132 @@ public Page maximizeWindow() {
     }
-    
-	/**
-	 * Switches focus of the WebDriver to a new window assuming there was only one
-	 * before. Finds the handle and passes that to overloaded switchToNewWindow()
-	 * method
-	 * <p>
-	 * <b>Preconditions:</b> Expects a new tab or window to have just been opened,
-	 * and for there to be only two.
-	 * 
-     * @param pageName the new page name to set the window to so it switches correctly
-     * @return String the window handle we are switching to
-	 * @throws InterruptedException if page doesn't load
+
+    /**
+     * Switches to a previously utilized window which still exists
+     * <p>
+     * <b>Preconditions:</b> Only one page exists for each page object
+     * @param pageName The name of the page to switch to
+     * @return The window handle of the switched-to window
+     * @throws NoSuchWindowException If no page of that name was utilized previously
      */
-	public static String switchToNewWindow(String pageName) throws InterruptedException {
-		String newHandle = null;
-		parentHandle = driver().getWindowHandle();
-		parentPage = PageManager.getPage().getName();
-		Set<String> handles = driver().getWindowHandles();
-		if (handles.size() == 1) {
-			var errorMessage = "Only one window is open, therefore we cannot switch to a new window. Please open a new window and try again.";
-			log.error(errorMessage);
-			throw new NoSuchWindowException(errorMessage);
+    public static String switchToExistingWindow(String pageName) {
+    	previousPageInfo = currentPageInfo;
+
+    	if(pages.containsKey(pageName)) {
+    		driver().switchTo().window(pages.get(pageName).getRight()[0]);
+    		setPage(pageName);
+    	}
+    	else {
+    		throw new NoSuchWindowException("No page of that name was used previously");
+    	}
+
+    	return driver().getWindowHandle();
+    }
+
+    /**
+     * Switches to the first found unvisited page and associates it with the page object
+     * associated with the passed page name.
+     * <p>
+     * <b>Preconditions:</b> Only one new window should have opened.
+     * @param pageName The page object to associate with the newly switched to page
+     * @return The window handle of the switched-to window
+     * @throws NoSuchWindowException if a new window cannot be switched to in the configured
+     * timeout.
+     */
+    public static String switchToNewWindow(String pageName) {
+    	previousPageInfo = currentPageInfo;
+    	ArrayList<String> allKnownHandles = new ArrayList<>();
+    	
+    	for(Map.Entry<String, Pair<Page, String[]>> entry : pages.entrySet()) {
+			allKnownHandles.add(entry.getValue().getRight()[0]);
 		}
-		if (parentHandle == null) {
-			var errorMessage = "Parent Window cannot be found. Please open a window and restart your test.";
-			log.error(errorMessage);
-			throw new NoSuchWindowException(errorMessage);
+
+		try {
+			FluentWait<WebDriver> wait = new FluentWait<>(driver())
+				       .withTimeout(Time.out().plusSeconds(30))
```

Comment:
This is a hidden amount of time. Every other timeout is configurable except this one. You are forcing a test to fail after a minimum of 31 seconds even if the test doesn't take that long. You are looking at this in this way because you are trying to solve your specific problem of a slow windows app. The solution to this is to up your timeout, not force everyone to up theirs.